crates/decals/src/decal.wgsl

```diff
@@ -21,7 +21,42 @@ struct VertexOutput {
 
 fn get_entity_primitive_mesh(loc: vec2<u32>, index: u32) -> u32 {
     var meshes = get_entity_gpu_primitives_mesh(loc);
-    return bitcast<u32>(meshes[index >> 2u][index & 3u]);
+    if index == 0u {
+        return bitcast<u32>(meshes[0][0]);
+    } else if index == 1u {
+        return bitcast<u32>(meshes[0][1]);
+    } else if index == 2u {
+        return bitcast<u32>(meshes[0][2]);
+    } else if index == 3u {
+        return bitcast<u32>(meshes[0][3]);
+    } else if index == 4u {
+        return bitcast<u32>(meshes[1][0]);
+    } else if index == 5u {
+        return bitcast<u32>(meshes[1][1]);
+    } else if index == 6u {
+        return bitcast<u32>(meshes[1][2]);
+    } else if index == 7u {
+        return bitcast<u32>(meshes[1][3]);
+    } else if index == 8u {
+        return bitcast<u32>(meshes[2][0]);
+    } else if index == 9u {
+        return bitcast<u32>(meshes[2][1]);
+    } else if index == 10u {
+        return bitcast<u32>(meshes[2][2]);
+    } else if index == 11u {
+        return bitcast<u32>(meshes[2][3]);
+    } else if index == 12u {
+        return bitcast<u32>(meshes[3][0]);
+    } else if index == 13u {
+        return bitcast<u32>(meshes[3][1]);
+    } else if index == 14u {
+        return bitcast<u32>(meshes[3][2]);
+    } else if index == 15u {
+        return bitcast<u32>(meshes[3][3]);
+    } else {
+        // unreachable
+        return 1000u;
+    }
```

Comment:
Hm, it looks like this code is used in multiple places. Maybe move it to a common include, so that even if this is necessary, at least they won't go out of sync.

Reply:
Yes.

The issue with that is that there wasn't really a common module, since the modules that are shared are either *before* the mesh functions, or they don't share the same layouts

Reply:
> Is it possible to do div / modulo (which it looks like the original code was doing)?

No, it is not the math, but the fact that dynamic pointer indexing don't work for non-reference types on matrices under tint, so we have to unroll and use constant indexing

Reply:
Additionally, I don't think the terrain isn't even used anywhere in the code, and this was something @Pombal mentioned in Barcelona when we first went through the code base

Reply:
Can you please add a comment on the shader explaning why it's done this way instead of the original?
And if we can find a way to deduplicate the code, even if creating a new common include, that would be great.

Reply:
Done